src/redux/books/booksSlice.js

```diff
@@ -1,42 +1,63 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import axios from 'axios';
 
 const initialState = {
-  books: [
-    {
-      item_id: 'item1',
-      title: 'The Great Gatsby',
-      author: 'John Smith',
-      category: 'Fiction',
-    },
-    {
-      item_id: 'item2',
-      title: 'Anna Karenina',
-      author: 'Leo Tolstoy',
-      category: 'Fiction',
-    },
-    {
-      item_id: 'item3',
-      title: 'The Selfish Gene',
-      author: 'Richard Dawkins',
-      category: 'Nonfiction',
-    },
-  ],
+  books: [],
 };
 
+const BASE_URL = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/LGxYf44oPI6RSiuKvX6C/books';
+
+export const fetchBooks = createAsyncThunk('books/fetchBooks', async () => {
+  try {
+    const response = await axios.get(BASE_URL);
+    const res = response.data;
+    const books = Object.keys(res).map((key) => ({
+      item_id: key,
+      ...res[key][0],
+    }));
+    return books;
+  } catch (error) {
+    return error.message;
+  }
+});
+
+export const removeBook = createAsyncThunk('books/removeBook', async (id) => {
+  try {
+    await axios.delete(`${BASE_URL}/${id}`);
+    return id;
+  } catch (error) {
+    return error.message;
+  }
+});
+
+export const addBook = createAsyncThunk('books/addBook', async (book) => {
+  try {
+    await axios.post(BASE_URL, book);
+    return book;
+  } catch (error) {
+    return error.message;
+  }
+});
+
```

Comment:
- [  ] Please instead of using try and catch it would be better to just return a promise instead, and verify that the book was added.